qiskit/quantum_info/operators/symplectic/clifford.py

```diff
@@ -110,109 +113,281 @@ def __array__(self, dtype=None):
     def __init__(self, data, validate=True):
         """Initialize an operator object."""
 
-        # Initialize from another Clifford by sharing the underlying
-        # StabilizerTable
+        # Initialize from another Clifford by sharing the data
         if isinstance(data, Clifford):
-            self._table = data._table
+            num_qubits = data.num_qubits
+            self.tableau = data.tableau
 
         # Initialize from ScalarOp as N-qubit identity discarding any global phase
         elif isinstance(data, ScalarOp):
             if not data.num_qubits or not data.is_unitary():
                 raise QiskitError("Can only initialize from N-qubit identity ScalarOp.")
-            self._table = StabilizerTable(np.eye(2 * data.num_qubits, dtype=bool))
+            num_qubits = data.num_qubits
+            self.tableau = np.fromfunction(
+                lambda i, j: i == j, (2 * num_qubits, 2 * num_qubits + 1)
+            ).astype(bool)
 
         # Initialize from a QuantumCircuit or Instruction object
         elif isinstance(data, (QuantumCircuit, Instruction)):
-            self._table = Clifford.from_circuit(data)._table
+            num_qubits = data.num_qubits
+            self.tableau = Clifford.from_circuit(data).tableau
 
+        # DEPRECATED: data is StabilizerTable
+        elif isinstance(data, StabilizerTable):
+            self.tableau = self._stack_table_phase(data.array, data.phase)
+            num_qubits = data.num_qubits
         # Initialize StabilizerTable directly from the data
         else:
-            self._table = StabilizerTable(data)
+            if isinstance(data, (list, np.ndarray)) and np.asarray(data, dtype=bool).ndim == 2:
+                data = np.asarray(data, dtype=bool)
+                if data.shape[0] == data.shape[1]:
+                    self.tableau = self._stack_table_phase(
+                        data, np.zeros(data.shape[0], dtype=bool)
+                    )
+                    num_qubits = data.shape[0] // 2
+                elif data.shape[0] + 1 == data.shape[1]:
+                    self.tableau = data
+                    num_qubits = data.shape[0] // 2
+                else:
+                    raise QiskitError("")
+            else:
+                n_paulis = len(data)
+                symp = self._from_label(data[0])
+                num_qubits = len(symp) // 2
+                tableau = np.zeros((n_paulis, len(symp)), dtype=bool)
+                tableau[0] = symp
+                for i in range(1, n_paulis):
+                    tableau[i] = self._from_label(data[i])
+                self.tableau = tableau
 
             # Validate table is a symplectic matrix
-            if validate and not Clifford._is_symplectic(self._table.array):
+            if validate and not Clifford._is_symplectic(self.tableau[:, :-1]):
```

Comment:
Consider defining a property and a setter for the symplectic part of the clifford tableau, say `symplectic`, instead of the previous `cliff.table.array` and not call `tableau[:, :-1]` (which appears several times in the code).

Reply:
Thanks. The properties are nice. The property name `symplectic` is well used in the literature? I know that we check the symplectic inner product in the method `_is_symplectic`, but I don't know whether Z components and X components called symplectic.

Reply:
The 2nx2n array in the clifford tableau is a symplectic binary matrix, see e.g. https://arxiv.org/abs/1406.2170 (eq. 3 and 4) and https://en.wikipedia.org/wiki/Symplectic_group for the definition.
But of course, it's possible to choose a different name if you like.

Reply:
Yeah, it's an element of symplectic group. Thank you for the explanation. Then, I prefer `Clifford.symplectic_matrix` (a bit long).

Reply:
Add in c1a8f98. It is more readable.